core/tests/test_apps.bitcoin.signtx_decred.py

```diff
@@ -0,0 +1,368 @@
+from apps.bitcoin.sign_tx import decred, helpers
+from apps.bitcoin.sign_tx.approvers import BasicApprover
+
+
+EMPTY_SERIALIZED = TxRequestSerializedType(serialized_tx=bytearray())
+
+coin_decred = coins.by_name("Decred")
+
+ptx1 = PrevTx(version=1, lock_time=0, inputs_count=2, outputs_count=1, extra_data_len=0)
+pinp1 = PrevInput(
+    script_sig=unhexlify(
+        "483045022072ba61305fe7cb542d142b8f3299a7b10f9ea61f6ffaab5dca8142601869d53c0221009a8027ed79eb3b9bc13577ac2853269323434558528c6b6a7e542be46e7e9a820141047a2d177c0f3626fc68c53610b0270fa6156181f46586c679ba6a88b34c6f4874686390b4d92e5769fbb89c8050b984f4ec0b257a0e5c4ff8bd3b035a51709503"
+    ),
+    prev_hash=unhexlify(
+        "c16a03f1cf8f99f6b5297ab614586cacec784c2d259af245909dedb0e39eddcf"
+    ),
+    prev_index=1,
+    sequence=0xFFFF_FFFF,
+)
+pinp2 = PrevInput(
+    script_sig=unhexlify(
+        "48304502200fd63adc8f6cb34359dc6cca9e5458d7ea50376cbd0a74514880735e6d1b8a4c0221008b6ead7fe5fbdab7319d6dfede3a0bc8e2a7c5b5a9301636d1de4aa31a3ee9b101410486ad608470d796236b003635718dfc07c0cac0cfc3bfc3079e4f491b0426f0676e6643a39198e8e7bdaffb94f4b49ea21baa107ec2e237368872836073668214"
+    ),
+    prev_hash=unhexlify(
+        "1ae39a2f8d59670c8fc61179148a8e61e039d0d9e8ab08610cb69b4a19453eaf"
+    ),
+    prev_index=1,
+    sequence=0xFFFF_FFFF,
+)
+pout1 = PrevOutput(
+    script_pubkey=unhexlify("76a91424a56db43cf6f2b02e838ea493f95d8d6047423188ac"),
+    amount=200000 + 200000 - 10000,
+    decred_script_version=0,
+)
+
+
+@unittest.skipUnless(not utils.BITCOIN_ONLY, "altcoin")
+class TestSignTxDecred(unittest.TestCase):
+    # pylint: disable=C0301
+
+    def test_one_one_fee(self):
+
+        inp1 = TxInput(
+            address_n=[44 | 0x80000000, 42 | 0x80000000, 0 | 0x80000000, 0, 0],
+            prev_hash=unhexlify(
+                "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+            ),
+            prev_index=0,
+            amount=390000,
+            multisig=None,
+            sequence=0xFFFF_FFFF,
+        )
+        out1 = TxOutput(
+            address="DsaHnKa418BeeQmyhpQEGG4cxGAPrneydfv",
+            amount=390000 - 10000,
+            script_type=OutputScriptType.PAYTOADDRESS,
+            multisig=None,
+        )
+        tx = SignTx(
+            coin_name="Decred", version=1, lock_time=0, inputs_count=1, outputs_count=1
+        )
+
+        messages = [
+            None,
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(request_index=0, tx_hash=None),
+                serialized=TxRequestSerializedType(
+                    serialized_tx=unhexlify("0100000001")
+                ),
+            ),
+            TxAckInput(tx=TxAckInputWrapper(input=inp1)),
+            TxRequest(
+                request_type=TXOUTPUT,
+                details=TxRequestDetailsType(request_index=0, tx_hash=None),
+                serialized=EMPTY_SERIALIZED,
```

Comment:
Regarding https://github.com/trezor/trezor-firmware/pull/1249/files#r528434214: The `serialized_tx` is not empty. The same also applies to the `TxRequest`s below.